core/templates/dev/head/domain/exploration/RuleObjectFactorySpec.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit test for RuleObjectFactory.
+ */
+import { TestBed } from '@angular/core/testing';
+import { RuleObjectFactory, Rule } from
+  'domain/exploration/RuleObjectFactory';
+
+describe('RuleObjectFactory', () => {
+  let ruleObjectFactory;
+  let ruleBackendDict: any;
+
+  beforeEach(() => {
+    ruleObjectFactory = TestBed.get(RuleObjectFactory);
+    ruleBackendDict = {
+      rule_type: 'rule_type_1',
+      inputs: 'input_1'
```

Comment:
`inputs` is supposed to be an array of dicts, so can you give the correct value here? (You can check class RuleSpec in state_domain.py for reference).